kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -357,12 +387,141 @@ verifyAxiomSentence sentence =
                 (freeVariables sentence)
                 (sentenceAxiomAttributes sentence)
         State.modify $ addAxiom verified attrs
+        case fromSentenceAxiom (attrs, verified) of
+            Right eq@Equation{left, argument} ->
+                when (needsVerification eq) $
+                    checkLHS eq left >> checkArg eq argument
+            Left (RequiresError _) ->
+                koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                    "RequiresError thrown during equation check"
+            Left (ArgumentError _) ->
+                koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                    "ArgumentError thrown during equation check"
+            Left (AntiLeftError _) ->
+                koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                    "AntiLeftError thrown during equation check"
+            Left (EnsuresError _) ->
+                koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                    "EnsuresError thrown during equation check"
```

Comment:
Please rewrite all of these error messages with terminology that will be familiar to the user. The best thing to do would be to add a `Pretty` instance for this error type, and present it with `koreFailWithLocations`.